src/Environment.php

```
@@ -297,72 +284,123 @@ public static function isProd(): bool {
-    }
-    static::$override = $callback;
-  }
-
-  /**
-   * Get the fallback environment type.
+   * @param bool $all
+   *   Whether to reset all settings.
    */
-  public static function fallback(): string {
-    return static::$fallback;
+  public static function reset(bool $all = FALSE): void {
```

PHPMD (2.15.0)

The method reset has a boolean flag argument $all, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)